src/main/java/frc/robot/subsystems/drive/states/LinearDriveState.java

```diff
@@ -0,0 +1,373 @@
+    Rotation2d theta = velocityAngle.minus(targetAngle);
+
+    // Project the velocity vector onto the speaker vector with ||u|| * cos(theta)
+    // Where u is the velocity vector
+    double velocityToGoal = velocity.norm() * theta.getCos();
+
+    return velocityToGoal;
+  }
+
+  @Override
+  public void periodic() {
+    Pose2d currentPose = drive.getPose();
+
+    // Project the goal pose backwards to find the phase 1 goal pose.
+    double phase1OffsetX =
+        kPhase2Distance * Math.cos(goalPose.getRotation().getRadians() + Math.PI);
+    double phase1OffsetY =
+        kPhase2Distance * Math.sin(goalPose.getRotation().getRadians() + Math.PI);
+    Pose2d phase1Pose =
+        new Pose2d(
+            goalPose.getX() + phase1OffsetX,
+            goalPose.getY() + phase1OffsetY,
+            goalPose.getRotation());
+
+    // Get distances to goal positions.
+    double distanceToGoal = currentPose.getTranslation().getDistance(goalPose.getTranslation());
+
+    // double distanceToPhase1Pose =
+    //     currentPose.getTranslation().getDistance(phase1Pose.getTranslation());
+
+    // Find if the robot is within the slice area to transition to phase 2.
+    Translation2d phase1PoseToGoal = goalPose.minus(phase1Pose).getTranslation();
+    Translation2d robotToGoal = goalPose.minus(currentPose).getTranslation();
+    double angleToTarget = getAngleBetweenVectors(robotToGoal, phase1PoseToGoal);
+
+    // Determine which pose to aim at. However, always use the distance to the final pose to compute
+    // the target speed.
+    Pose2d currentGoalPose = phase1Pose;
+    if (distanceToGoal < kPhase2Distance || Math.abs(angleToTarget) < kPhase2Angle) {
+      currentGoalPose = goalPose;
+    }
+
+    // We only exit this state when the phase 2 pose has been achieved.
+    if (distanceToGoal < lineupErrorMargin) {
+      if (drive.isDesiredLocationReef()) {
+        drive.fireTrigger(DriveTrigger.BeginLineup);
+      } else {
+        drive.fireTrigger(DriveTrigger.CancelLinear);
+      }
+      return;
+    }
+
+    // HEADING
+    double headingError = currentPose.getRotation().minus(goalPose.getRotation()).getRadians();
+    double headingVelocity =
+        headingController.calculate(
+            currentPose.getRotation().getRadians(), goalPose.getRotation().getRadians());
+
+    // double driveVelocityScalar = driveController.calculate(distanceToGoal, 0.0);
+
+    ChassisSpeeds chassisSpeeds =
+        ChassisSpeeds.fromRobotRelativeSpeeds(drive.getChassisSpeeds(), currentPose.getRotation());
+    Vector<N2> velocity =
+        VecBuilder.fill(chassisSpeeds.vxMetersPerSecond, chassisSpeeds.vyMetersPerSecond);
+
+    double velocityToGoal =
+        findVelocityTowardPoint(
+            velocity, currentPose.getTranslation(), currentGoalPose.getTranslation());
+
+    double driveVelocityScalar =
+        driveProfile.calculate(
+                0.02,
+                new State(distanceToGoal, -Math.abs(velocityToGoal)),
```

Comment:
Why is this `-Math.abs(velocityToGoal)`? Like, why is the negative sign on the outside of this (we'll always have a negative initial velocity with this)?